pkg/api/plugin.go

```diff
@@ -121,4 +123,7 @@ type GenevaActions interface {
 
 	// GetControlPlanePods fetches a consolidated list of the control plane pods in the cluster
 	GetControlPlanePods(ctx context.Context, oc *OpenShiftManagedCluster) ([]byte, error)
+
+	// ReimageAndRestart reimages and restarts a virtual machine within a scale set in the cluster.
+	ReimageAndRestart(ctx context.Context, oc *OpenShiftManagedCluster, vmScaleSetName string, instanceId string) *PluginError
```

Comment:
should accept a hostname, not vmscalesetname/instanceid

Reply:
I think you can return a straight `error` here - don't worry about PluginError consts

Reply:
done